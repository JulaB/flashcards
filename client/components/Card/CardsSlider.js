import React from 'react';
import PropTypes from 'prop-types';
import Slider from 'components/Shared/Slider/Slider';
import Card from 'components/Card/Card';

const CardsSlider = ({ cards }) => (
  <Slider>
    {cards.map(card => <Card {...card.attributes} key={card.id} />)}
  </Slider>
);

CardsSlider.propTypes = {
  cards: PropTypes.arrayOf(PropTypes.object),
};

CardsSlider.defaultProps = {
  cards: [],
};
export default CardsSlider;
