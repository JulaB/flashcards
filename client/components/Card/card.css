.card {
  width: 100%;
  min-height: 200px;

  &__term,
  &__definition {
    border: 1px solid var(--color-accent);
    width: inherit;
    height: inherit;
    display: flex;
    padding: 10px;
    align-items: center;
    justify-content: center;
  }

  &__term {
    font-size: var(--text-xl);
  }

  &:focus,
  &:hover {
    .card__term,
    .card__definition {
      border: 1px solid var(--color-primary-dark);
    }
  }
}
