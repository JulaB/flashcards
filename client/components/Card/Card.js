import React from 'react';
import PropTypes from 'prop-types';
import FlipBox from 'components/Shared/FlipBox/FlipBox';

import './card.css';

const Card = ({ term, definition }) => (
  <FlipBox className="card">
    <div className="card__term">{term}</div>
    <div className="card__definition" >{definition}</div>
  </FlipBox>
);

Card.propTypes = {
  term: PropTypes.string.isRequired,
  definition: PropTypes.string,
};

Card.defaultProps = {
  definition: '',
};

export default Card;
