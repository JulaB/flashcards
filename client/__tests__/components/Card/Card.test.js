import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Card from 'components/Card/Card';

describe('<Card />', () => {
  it('renders component', () => {
    const wrapper = shallow(<Card term="test" definition="some definition" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
