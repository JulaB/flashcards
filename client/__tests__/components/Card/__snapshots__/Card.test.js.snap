// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders component 1`] = `
<FlipBox
  className="card"
>
  <div
    className="card__term"
  >
    test
  </div>
  <div
    className="card__definition"
  >
    some definition
  </div>
</FlipBox>
`;
